packages/server/src/routes/provider/alert/create.svelte

```diff
@@ -546,36 +587,105 @@
+          on:click={() => {
+            isIconSelectorOpen = false;
+            isCustomIconSelectorOpen = false;
+          }} />
       </header>
-      <section class="modal-card-body">
-        <div class="icon-choiced-container">
-          <div class="icon-choices">
-            {#each Object.values(icons) as icon}
-              <div class="icon-container">
-                <button
-                  class="button is-large"
-                  class:has-background-info={tempIcon === icon}
-                  on:click={() => (tempIcon = icon)}>
-                  <span class="icon is-large">
-                    <i class={`${icon} fa-2x`} />
-                  </span>
-                </button>
+      {#if isCustomIconSelectorOpen}
+        <section class="modal-card-body">
+          <div class="icon-choiced-container">
+            {#if iconSearchResults}
+              <div class="icon-choices">
+                {#each iconSearchResults as icon}
+                  <div class="icon-container">
+                    <button
+                      class="button is-large"
+                      class:has-background-info={tempIcon === icon}
+                      on:click={() => {
+                        tempIcon = icon;
+                        icons[icon] = icon;
+                      }}>
+                      <span class="icon is-large">
+                        <i class={`${icon} fa-2x`} />
+                      </span>
+                    </button>
+                  </div>
+                {/each}
+              </div>
+            {:else}
+              <div>
+                <h5>No results found</h5>
               </div>
-            {/each}
+            {/if}
           </div>
-        </div>
-      </section>
-      <footer class="modal-card-foot">
-        <button
-          class="button is-success"
-          on:click={() => {
-            alertIcon = tempIcon;
-            isIconSelectorOpen = false;
-          }}>
-          Save changes
-        </button>
-      </footer>
+          <div class="control icon-search-bar">
+            <input
+              class="input"
+              type="text"
+              placeholder="Search for an icon..."
+              bind:value={iconSearchTerm} />
+          </div>
+          <div class="control">
+            <button class="button" on:click={searchForCustomIcons}>
+              Search
+            </button>
+          </div>
+        </section>
+        <footer class="modal-card-foot">
+          <button
+            class="button is-success"
+            on:click={() => {
+              alertIcon = tempIcon;
+              isIconSelectorOpen = false;
+            }}>
+            Save changes
+          </button>
+          <button
+            class="button is-info"
+            on:click={() => {
+              isCustomIconSelectorOpen = false;
+            }}>
+            Back
+          </button>
+        </footer>
```

Comment:
Can we move this footer outside the conditional so that the pre-set icons and searched icons can share it to reduce duplication? We could just have a conditional on the 2nd button that handles the button text and action depending on whether they're looking at pre-set or searching icons.